tests/spark/rflx-tlv_tests.adb

```diff
@@ -263,6 +263,61 @@ package body RFLX.TLV_Tests is
       Free_Bytes_Ptr (Buffer);
    end Test_Generating_TLV_Error;
 
+   procedure Test_Read_Write_Reset (T : in out AUnit.Test_Cases.Test_Case'Class) with
+     SPARK_Mode, Pre => True
+   is
+      pragma Unreferenced (T);
+      Buffer  : RFLX_Builtin_Types.Bytes_Ptr := new RFLX_Builtin_Types.Bytes'(1, 0, 4, 0, 0, 0, 0);
+      Context : TLV.Message.Context;
+   begin
+      TLV.Message.Initialize (Context, Buffer);
+      TLV.Message.Verify_Message (Context);
+
+      Assert (TLV.Message.Structural_Valid_Message (Context), "Structural invalid message after initialization");
+
+      declare
+         pragma Warnings (Off, "subprogram ""Read"" has no effect");
+         procedure Read (Buffer : RFLX_Builtin_Types.Bytes) is
+         begin
+            Assert (Buffer, (1, 0, 4, 0, 0, 0, 0), "Invalid binary representation");
+         end Read;
+         pragma Warnings (On, "subprogram ""Read"" has no effect");
+         pragma Warnings (Off, "subprogram ""Message_Read"" has no effect");
+         procedure Message_Read is new TLV.Message.Read (Read);
+         pragma Warnings (On, "subprogram ""Message_Read"" has no effect");
+      begin
+         Message_Read (Context);
+      end;
+
+      Assert (TLV.Message.Structural_Valid_Message (Context), "Structural invalid message after reading");
+
+      TLV.Message.Reset (Context);
+
+      Assert (not TLV.Message.Structural_Valid_Message (Context), "Structural valid message after reset");
+
+      declare
+         procedure Write (Buffer : out RFLX_Builtin_Types.Bytes) is
+         begin
+            Assert (Buffer'Length = 7, "Invalid buffer length");
+            Buffer := (1, 0, 4, 0, 0, 0, 0);
```

Comment:
Wouldn't it be better to choose a message different from the original one?

Reply:
You are right. The test case is improved by 58117bd.